### PR TITLE
Fixes awslambda policy management

### PR DESCRIPTION
**Pull request by bryanlalexander**
- adds support for multiple policy statements
- adds support for RevisionId parameter
- adds support for lambda/RemovePermission
- adds State parameter to lambda/GetFunctionConfiguration (always returns Active)

Most of these changes were required in order to more closely mimic the AWS API and resolve issues when using moto[server] with terraform.

### CODE REVIEW

**Comment by brady-gsa** [comment]
@bblommers @spulec @JackDanger

Can you review this one as well?

**Comment by bryanlalexander** [comment]
Removed [WIP] from title, PR is ready for initial review

File: moto/awslambda/policy.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        self.parent = parent
+
+    def __repr__(self):
+        return json.dumps(self.get_policy())
+
+    def wire_format(self):
+        return json.dumps(
+            {
+                "Policy": json.dumps(
+                    {
```
Comment (bblommers) [inline moto/awslambda/policy.py]: Can we re-use the get-policy method here?
It looks like the JSON is exactly the same

File: tests/test_awslambda/test_policy.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+from moto.awslambda.policy import Policy
+
+
+class MockLambdaFunction:
+    def __init__(self, arn):
+        self.function_arn = arn
+        self.policy = None
+
+
+class TC:
```
Comment (bblommers) [inline tests/test_awslambda/test_policy.py]: Can you rework the tests to be more inline with the rest of Moto?
Typically we just have separate methods per test case, i.e.:

def test_add_statement():
    # setup
    # add statement
    # assert ...

File: moto/awslambda/policy.py (hunk 1 above)
Comment (bryanlalexander) [reply moto/awslambda/policy.py]: wire_format returns JSON and the `Policy` must be escaped JSON so it's somewhat different, however, I agree I was looking at that the other day. We'll take a look and see if we can refactor it a bit.

File: tests/test_awslambda/test_policy.py (hunk 2 above)
Comment (bryanlalexander) [reply tests/test_awslambda/test_policy.py]: We'll get it swapped to one execution of all possible policy options instead of a table driven format.

**Comment by bryanlalexander** [comment]
@bblommers @brady-gsa had the cycles and got those fixed, if you wouldn't mind taking another look.

**Comment by bblommers** [comment]
Looks mostly good to me @bryanlalexander. 

One minor niggle: Can you remove the TestPolicy-class? 
The test-method itself can be made global - as long as the name starts with _test_, Nosetests will recognize it.

**Comment by brady-gsa** [comment]
@bblommers we use assertions in our tests so if we remove the class we'd do something like the below. Do you still want us to remove it?
```
tc = unittest.TestCase('__init__')
tc.assertEquals(a, b)
```
We also had trouble running individual tests with nosetest when there was no test class. Can you give a little background why it was decided not to use test classes?

**Comment by bblommers** [comment]
I don't know why this structure was decided, that was well before my time. I'm just trying to keep things consistent :) 

Other tests typically use _sure_ for assertions:
```
import sure # noqa
a.should.equal(b)
```
E.g.: https://github.com/spulec/moto/blob/master/tests/test_awslambda/test_lambda.py#L185

**Comment by brady-gsa** [comment]
@bblommers we can look into using sure. Do you know how to run a single test with nosetest when there is no test class?

**Comment by bblommers** [comment]
@brady-gsa  Out the top of my head you can simply use the file as a command line argument:
```
nosetests -v tests/test_tests/test_awslambda/test_policy.py
nosetests -v tests/test_tests/test_awslambda/test_policy.py:test_method_name
```
Might have to run this in a virtualenv

**Comment by brady-gsa** [comment]
> @brady-gsa Out the top of my head you can simply use the file as a command line argument:
> 
> ```
> nosetests -v tests/test_tests/test_awslambda/test_policy.py
> nosetests -v tests/test_tests/test_awslambda/test_policy.py:test_method_name
> ```
> 
> Might have to run this in a virtualenv

Thanks @bblommers, specifying the file worked fine - I thought we had tried that. I updated the tests to use sure.

One thing that's a little strange about sure is that PyCharm will throw errors such as the below. Do you know of a solution for that?

> "unresolved attribute should for class dict"

**Comment by bblommers** [comment]
```
unresolved attribute should for class dict
```
That probably means that PyCharm hasn't indexed the sure-package, or doesn't know it's part of the project.
```
unused import statement sure
```
That's expected - on import it will do some magic to add the 'should'-attribute to dicts, but it's not referred anywhere after that.
Most tests will have _# noqa_ after the import to ignore that warning.

Thank you for your PR!

**Pull request merged**